src/App.tsx

```diff
@@ -1,27 +1,84 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [clickedButton, setClickedButton] = useState('');
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const loadGoods = async () => {
+    try {
+      switch (clickedButton) {
+        case 'all':
+          setGoods(await getAll());
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+          return;
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+        case 'five':
+          setGoods(await get5First());
 
-    <GoodsList goods={[]} />
-  </div>
-);
+          return;
+
+        case 'red':
+          setGoods(await getRedGoods());
+
+          return;
+
+        default:
+          setGoods([]);
+
+          return;
+      }
+    } catch (error) {
+      setGoods([]);
+    }
+  };
+
+  useEffect(() => {
+    loadGoods();
+  }, [clickedButton]);
+
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
+
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={() => {
+          setClickedButton('all');
+          loadGoods();
```

Comment:
I think it's better to create 3 separate functions in order of trying to handle everything in one. 
It would make your code cleaner and more understandable. It's an interesting solution, but loadGoods function looks too big and complicated

Reply:
done